File: scripts/build_pages_publish.py
from __future__ import annotations

import shutil
from pathlib import Path


ROOT_DIR = Path(__file__).resolve().parents[1]
OUTPUT_DIR = ROOT_DIR / '.pages-build'

ROOT_FILES = (
  'index.html',
  'edit.html',
  'documents.html',
  'styles.css',
  'structure.yaml',
)

ROOT_DIRS = (
  'js',
)

DATA_DIRS = (
  'people',
  'text_processing',
)

TEXT_DOCUMENTS_DIR = Path('data') / 'sources' / 'text_documents'


def reset_output_dir() -> None:
  if OUTPUT_DIR.exists():
    shutil.rmtree(OUTPUT_DIR)
  OUTPUT_DIR.mkdir(parents=True, exist_ok=True)


def copy_file(relative_path: str | Path) -> None:
  source = ROOT_DIR / relative_path
  target = OUTPUT_DIR / relative_path
  target.parent.mkdir(parents=True, exist_ok=True)
  shutil.copy2(source, target)


def copy_tree(relative_path: str | Path) -> None:
  source = ROOT_DIR / relative_path
  target = OUTPUT_DIR / relative_path
  shutil.copytree(source, target)


def main() -> None:
  reset_output_dir()

  for file_name in ROOT_FILES:
    copy_file(file_name)

  for dir_name in ROOT_DIRS:
    copy_tree(dir_name)

  for dir_name in DATA_DIRS:
    copy_tree(Path('data') / dir_name)

  if (ROOT_DIR / TEXT_DOCUMENTS_DIR).exists():
    copy_tree(TEXT_DOCUMENTS_DIR)

  print(f'Pages publish directory ready: {OUTPUT_DIR}')


if __name__ == '__main__':
  main()

File: scripts/build-pages-publish.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const rootDir = path.resolve(__dirname, "..");
const outputDir = path.join(rootDir, ".pages-build");

const rootFiles = [
  "index.html",
  "edit.html",
  "documents.html",
  "styles.css",
  "structure.yaml",
];

const rootDirs = [
  "js",
];

const dataDirs = [
  path.join("data", "people"),
  path.join("data", "text_processing"),
];

const textDocumentsDir = path.join("data", "sources", "text_documents");

function resetOutputDir() {
  fs.rmSync(outputDir, { recursive: true, force: true });
  fs.mkdirSync(outputDir, { recursive: true });
}

function copyFile(relativePath) {
  const source = path.join(rootDir, relativePath);
  const target = path.join(outputDir, relativePath);
  fs.mkdirSync(path.dirname(target), { recursive: true });
  fs.copyFileSync(source, target);
}

function copyDir(relativePath) {
  const source = path.join(rootDir, relativePath);
  const target = path.join(outputDir, relativePath);
  fs.cpSync(source, target, { recursive: true });
}

resetOutputDir();

for (const fileName of rootFiles) {
  copyFile(fileName);
}

for (const dirName of rootDirs) {
  copyDir(dirName);
}

for (const dirName of dataDirs) {
  copyDir(dirName);
}

if (fs.existsSync(path.join(rootDir, textDocumentsDir))) {
  copyDir(textDocumentsDir);
}

console.log(`Pages publish directory ready: ${outputDir}`);
